web/vibe/web/common.d

```diff
@@ -479,6 +475,36 @@ package struct MethodAttribute
 	alias data this;
 }
 
+///
+public struct ResultSerializer(alias ST, alias DT, string ContentType)
+{
+	alias content_type = ContentType;
+	alias serialize = ST;
+	alias deserialize = DT;
+}
+
+///
+package alias DefaultSerializerT (FuncRetT) =
+	ResultSerializer!(
+		function (FuncRetT rev_val){
+			return rev_val.serializeToJsonString();
```

Comment:
That is a good idea, and I will change the interface to accept a new output parameter, however I don't plan to restrict the interface to only accept OutputRanges. The client of the serializiation code might decide to use a custom output interface other than the OutputRange and the same client might also provide serialization UDAs for that specific custom output interface. Would there be any benefit restricting the type?

Reply:
Since the web/common.d code provides an object that needs to work for all user serializers, don't really see how to make this work without some form of common interface, and an output range is just the most simple form of that. But of course, since this would be a template parameter, the operations that the concrete type supports are not restricted, and user code could still make use of that.

Reply:
I could check those UDA's in the getInterfaceValidationError method, but I would need to iterate through all the functions and all the UDA's and then check a certain parameter. Would you prefer that?

Reply:
I don't even think that we need any explicit checks, just documenting it for `@resultSerializer` should be enough (would also be nice to have very a brief `unittest` example there).